tests/test_monitor.py

```diff
@@ -0,0 +1,217 @@
+
+
+@pytest.fixture
+def datamodel_test_instance():
+    """Test fixture for a DataModel object for use in testing BaseMonitor."""
+    class DataModelTestObject(BaseDataModel):
+        primary_key = 'a'
+
+        def get_new_data(self):
+            return NEW_DATA
+
+    return DataModelTestObject
+
+
+@pytest.fixture(
+    params=[NOTIFICATIONS, OUTPUT_DIR, OUTPUT_PATH, NAME, SUBPLOTS, LABELS, SIMPLE_SCATTER, SIMPLE_LINE, SIMPLE_IMAGE]
+)
+def monitor_test_instance(datamodel_test_instance, request):
+    """Test fixture for a Monitor object for use in testing. This fixture is parametrized with different attribute
+    configurations that affect Monitor behavior.
+
+    Note: request is a special pytest argument.
+    """
+    # Create a simple implementation of the BaseMonitor
+    class MonitorTestObject(BaseMonitor):
+        data_model = datamodel_test_instance
+        notification_settings = request.param.get('notification_settings', None)
+        output = request.param.get('output', None)
+        name = request.param.get('name', None)
+        subplots = request.param.get('subplots', False)
+        subplot_layout = request.param.get('subplot_layout', None)
+        labels = request.param.get('labels', None)
+        plottype = request.param.get('plottype', None)
+        x = request.param.get('x', None)
+        y = request.param.get('y', None)
+        z = request.param.get('z', None)
+
+        def get_data(self):
+            return self.model.new_data
+
+        def track(self):
+            return self.data.c * 2
+
+        def set_notification(self):
+            return 'notification'
+
+        def find_outliers(self):
+            return self.data.c >= 8
+
+    monitor_test_instance = MonitorTestObject()
+
+    # Yield that Monitor object
+    yield monitor_test_instance
+
+    # Post-test cleanup
+    if monitor_test_instance.model.model:  # Drop any data tables created during testing
+        monitor_test_instance.model.model.drop_table()
+
+    if monitor_test_instance.results_table is not None:  # Drop any results tables created during testing
+        monitor_test_instance._table.drop_table()
+
+    if os.path.exists(monitor_test_instance.output):  # Remove output files created during testing.
+        os.remove(monitor_test_instance.output)
+
+
+class TestMonitor:
+    """Test class for testing the monitorframe BaseMonitor."""
+    def test_monior_init(self, monitor_test_instance):
+        """Test that attributes are created or set correctly for each configuration."""
+        assert monitor_test_instance.model.new_data is not None  # New data should be retrieved by default
+
+        # Check that a table object is created and that the name is set correctly
+        assert (
+                monitor_test_instance._table is not None and
+                monitor_test_instance._table._meta.table_name == monitor_test_instance.__class__.__name__
+        )
```

Comment:
for these I like to add a printout message. This helps with Travis too. For example:

`assert a == b, "a which is {} is not equal to b which is {}".fromat(a, b)`

That way you can say what you tested for and print out any relevant information when the assertion fails. Not sure how useful this is here though.

Reply:
@robelgeda Turns out that this is redundant behavior if you're using `pytest`. In the default reports for `pytest` if the comparison fails there will be a line under the test that fails that "explains" why the assertion failed (with values included)